x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -92,9 +92,24 @@ public Literal visitBooleanValue(EsqlBaseParser.BooleanValueContext ctx) {
 
     @Override
     public Literal visitDecimalValue(EsqlBaseParser.DecimalValueContext ctx) {
+        boolean minus = false;
```

Comment:
I'd like this code here and the one in `visitIntegerLiteral` to be extracted (at least in part) to a dedicated method. Much of the code is duplicated atm.